ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -0,0 +1,218 @@
+        self.status = SQLSourceStatus()
+
+        self.metadata_config = metadata_config
+        self.metadata = OpenMetadata(metadata_config)
+
+        self.service = self.metadata.get_service_or_create(
+            entity=DatabaseService, config=config
+        )
+        self.engine: Engine = get_connection(self.service_connection)
+        self.test_connection()
+
+        self._session = None  # We will instantiate this just if needed
+        self._connection = None  # Lazy init as well
+        self.data_profiler = None
+        self.data_models = {}
+        self.table_constraints = None
+        self.database_source_state = set()
+        self.profile_date = datetime.now()
+        super().__init__()
+
+    def prepare(self):
+        self._parse_data_model()
+
+    @classmethod
+    def create(cls, config_dict: dict, metadata_config: OpenMetadataConnection):
+        pass
+
+    def standardize_schema_table_names(
+        self, schema: str, table: str
+    ) -> Tuple[str, str]:
+        return schema, table
+
+    def fetch_sample_data(self, schema: str, table: str) -> Optional[TableData]:
+        """
+        Get some sample data from the source to be added
+        to the Table Entities
+        """
+        try:
+            query = self.source_config.sampleDataQuery.format(schema, table)
+            logger.info(query)
+            results = self.connection.execute(query)
+            cols = []
+            for col in results.keys():
+                cols.append(col)
+            rows = []
+            for res in results:
+                row = list(res)
+                rows.append(row)
+            return TableData(columns=cols, rows=rows)
+        # Catch any errors and continue the ingestion
+        except Exception as err:  # pylint: disable=broad-except
+            logger.debug(traceback.format_exc())
+            logger.error(f"Failed to generate sample data for {table} - {err}")
+        return None
+
+    def get_databases(self) -> Iterable[Inspector]:
+        yield inspect(self.engine)
+
+    def get_schemas(self, inspector: Inspector) -> str:
+        for schema in inspector.get_schema_names():
+            yield schema
+
+    def _get_table_description(
+        self, schema: str, table_name: str, inspector: Inspector
+    ) -> str:
+        description = None
+        try:
+            table_info: dict = inspector.get_table_comment(table_name, schema)
+        # Catch any exception without breaking the ingestion
+        except Exception as err:  # pylint: disable=broad-except
+            logger.warning(f"Table Description Error : {str(err)}")
+        else:
+            description = table_info["text"]
+        return description
+
+    def _is_partition(self, table_name: str, schema: str, inspector: Inspector) -> bool:
+        self.inspector = inspector
+        return False
+
+    def get_table_names(self, schema: str, inspector: Inspector) -> Optional[List[str]]:
```

Comment:
The typing here does not look ok. We are returning an iterable of Tuple[str, str]